src/app/components/core/asset-readings/readings-graph/readings-graph.component.ts

```diff
@@ -968,17 +974,28 @@ export class ReadingsGraphComponent implements OnDestroy {
       });
   }
 
-  showBackwardReadingsGraphOrSummary() {
-    this.backwardReadingCounter++;
-    this.showReadingsGraphOrSummary();
-  }
+  moveInReadingsGraph(move: string) {
+    if(this.backwardReadingCounter === 0) {
+      this.mostRecentReadingTimestamp = this.timestamps[this.timestamps.length - 1];
+    }
+    if (move === "back") {
+      this.backwardReadingCounter++;
+    }
+    else {
+      this.backwardReadingCounter--;
+    }
 
-  showForwardReadingsGraphOrSummary() {
-    this.backwardReadingCounter--;
-    this.showReadingsGraphOrSummary();
+    if (!this.fromMostRecent) {
+      this.showReadingsGraph();
+      return;
+    }
+    let mostRecentReadingTimestampDate = new Date(this.mostRecentReadingTimestamp);
+    let prev = new Date(mostRecentReadingTimestampDate.valueOf() - this.backwardReadingCounter * this.optedTime * 1000);
+    let previous_ts = (moment(prev.valueOf()).format('YYYY-MM-DD HH:mm:ss.SSS')) + this.mostRecentReadingTimestamp.slice(-3); // send previous_ts upto 6 digits i.e. microsecond precision
```

Comment:
Please explain more about `previous_ts` format and its calculation.
While testing it, I'm getting it as `previous_ts = 2024-05-30%2001:52:39.934951`, here date it clear to me and microsecond is at last i.e. `934951` but what is 2001 ?

Reply:
We are using `YYYY-MM-DD HH:mm:ss.SSS` timestamp format for previous_ts, here `%20` is for space and `01` is hours